### PR TITLE
LAB-714: OpenAI-compat translation drops multimodal + array-form system content

### LINKED ISSUE #98: OpenAI compat translation drops multimodal + array-form system content in both directions
## Problem

Request translation drops or mangles non-plain-text content in both directions. Found by the LAB-708 OpenAI-surface audit; refs @ `ce0d44c`.

1. **O→A: system messages with array content are dropped entirely.** `translate_openai_to_anthropic` only accepts `content.as_str()` for `role: "system"` (`src/main.rs:7524`). OpenAI permits `content: [{"type":"text","text":...}]` on system messages; such a message vanishes silently.
2. **O→A: `image_url` parts are passed through untranslated.** Standard (non-tool) messages clone `content` as-is (`src/main.rs:7632-7635`). OpenAI's `{"type":"image_url","image_url":{"url":...}}` is not an Anthropic block type → upstream 400 on the whole request. Should translate to Anthropic `{"type":"image","source":{"type":"url"|"base64",...}}` (data: URLs → base64 source).
3. **A→O: image blocks are dropped silently.** `translate_anthropic_request_to_openai` filters user content to `type == "text"` only (`src/main.rs:8120-8127`). An Anthropic request with images routed to an OpenAI endpoint loses them without warning — worst case an empty `content: ""` user message.

## Fix sketch

- System: accept array form, join text parts (same pattern as the `tool` role handling at `src/main.rs:7536-7549`).
- Images O→A: map `image_url` → Anthropic `image` block (base64 for `data:` URLs, `url` source otherwise).
- Images A→O: map Anthropic `image` blocks → `image_url` parts; where an upstream can't take images, failing loudly (400 from the proxy) beats silent dropping.

Tracked in Multica: LAB-708 audit follow-up.

### PR DESCRIPTION
**Pull request by 27Bslash6**
Closes #98 (LAB-714).

Three silent-drop defects in the OpenAI-compat request translation, both directions:

1. **O→A: array-form system content dropped.** `translate_openai_to_anthropic` only read `content.as_str()` for `role: "system"`, so OpenAI's legal `content: [{"type":"text","text":...}]` form vanished silently. Now uses the same string-or-array extraction as the existing `tool`-role handler.
2. **O→A: `image_url` parts passed through untranslated.** Standard messages cloned `content` as-is, so an OpenAI `{"type":"image_url",...}` part reached Anthropic verbatim and 400'd the whole request. Now translated to Anthropic `image` blocks: `data:` URLs unpack into `base64` sources, other URLs become `url` sources. Malformed data URLs fall through to a `url` source so upstream rejects them with a real error instead of us dropping them.
3. **A→O: image blocks filtered out silently.** `translate_anthropic_request_to_openai` kept only `type == "text"` blocks in user content (worst case an empty `content: ""` message) — in both the plain-user path and the tool_result-leftover path. Both now route through a shared helper that emits OpenAI `image_url` parts; text-only arrays keep the plain-string content form, so existing behavior is unchanged when no images are present.

## Tests

Five new regression tests: array-form system content, O→A image translation (data: and https URLs), A→O image translation (base64 and url sources), text-only-array-stays-string invariant, and image-beside-tool_result survival. Full suite: 478 passed, 0 failed. `cargo fmt --check` and `clippy -Dwarnings` clean.

## Summary by CodeRabbit

* **New Features**
  * Improved request translation for image content, including correct handling of URL and base64 (data URL) sources.
  * Added support for system messages provided as multiple text parts.
  * Preserved mixed text, image, and tool-result content when translating requests.

* **Bug Fixes**
  * When translation fails during fallback routing, the app now returns a clear HTTP 400 `invalid_request_error` instead of retrying or rotating endpoints.
  * Better error surfacing for unsupported or malformed image/content blocks.

* **Tests**
  * Expanded coverage for translation edge cases and stricter assertions for stop-sequence formatting.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -8164,7 +8249,86 @@ fn reverse_map_stop_reason(reason: &str) -> &'static str {
...
+        serde_json::Value::Array(parts)
+    } else {
+        let text: String = parts
+            .iter()
+            .filter_map(|p| p.get("text").and_then(|t| t.as_str()))
+            .collect::<Vec<_>>()
+            .join("");
+        serde_json::Value::String(text)
+    })
+}
```
[the remark this answers is not in the document]
Comment (27Bslash6) [reply src/main.rs:8330]: Fixed in 8ad61ae — the wildcard arm now errors with the block type named (mirroring the image-source arm); untyped blocks stay ignored. Regression test added: a `document` block beside text fails the whole translation instead of being dropped.

**Comment by 27Bslash6** [comment]
@coderabbitai review
